### PR TITLE
Retain voxel shape information while floating ore exists.

### PR DESCRIPTION
**Pull request by alex-davidson**
MyFloatingObject entities still collide with voxel objects when grids and
players are out of range. If shape information has been discarded this
will cause expensive RequestShapeBlocking calls to happen frequently on the
simulation thread, with a noticeable effect on sim speed.

refs issue #376

### CODE REVIEW

**Comment by paradineai** [comment]
What will this do for processor, ram, and hdd use on a computer.

**Comment by alex-davidson** [comment]
Should not affect HDD use.

A few bytes extra memory usage per asteroid, plus a couple of kilobytes (max) per floating object near that asteroid (for 256 object limit, upper limit should be 1-2KB per 'roid depending on 32-bit vs 64-bit, I think).

Asteroid shapes will be kept in memory longer than previously, ie. until all nearby floating objects have despawned. I don't know how much memory one of these consumes but it should only affect you if you've run into this bug. If you're not mining asteroids, there should be no floating objects, and the shape should be evicted as normal.

Processor usage under normal circumstances should not change. Processor usage when the player is out of range of an asteroid with floating objects nearby should be a lot lower and gameplay a lot smoother.

tl;dr: if this bug has never affected you, the game's resource usage should not change noticeably. If this bug did affect you, RAM usage might increase slightly.

**Comment by MeusRex** [comment]
+1 (This is really necessary.)

After some heavy drilling on a remote asteroid my UPS drops from a constant 1.0 to 0.5-0.3 the moment I leave the asteroids immediate area. And once I create enough floating objects to force delete the ones close to the faraway asteroid the ups return to 1.0.

**Comment by CheeseJedi** [comment]
Wow - if this works, then awesome work!

**Comment by zrisher** [comment]
:+1:, keen please incorporate.

**Comment by kurevin** [comment]
+1 to this one if still applicable

**Comment by t3hk0d3** [comment]
+1
